docs/develop/optimizations.rst

```diff
@@ -0,0 +1,52 @@
+.. _optimizations:
+
+Optimizations
+=============
+
+MicroPython uses several optimizations to save RAM but also ensure the efficient
+execution of programs. This chapter discusses some of these optimizations.
+
+.. note::
+   :ref:`qstr` and :ref:`maps` details other optimizations on strings and 
+   dictionaries.
+
+
+Frozen bytecode
+----------------
+
+For efficiency, it is recommended that we avoid RAM or heap storage
+as much as possible. MicroPython code in scripts is stored on the heap during run-time interpreter execution.
+
+As an optimization, MicroPython code can be cross-compiled into bytecode
+that is stored and executed from flash instead of the heap. This leads to less
+RAM usage and fragmentation. Therefore precompiling application code
+into frozen bytecode will avoid unnecessary heap allocations
+
+Variables
+----------
+
+MicroPython processes local and global variables differently. Global variables
+are stored and looked up from a global dictionary that is allocated on the heap.
+Local variables on the other hand are stored on the stack which makes their access
+more efficient.
+
+The length of variable names also affects how much RAM is used as identifiers
```

Comment:
"of global variable names"  (local variable names are lost in the compilation process)